centreon/www/include/monitoring/recurrentDowntime/formDowntime.php

```diff
@@ -40,6 +40,60 @@
     exit();
 }
 
+/**
+ * GetUserAclAllowedResources returns allowed resources for the user regarding resource type
+ *
+ * @param CentreonACL $userAcl
+ * @param string $resourceType
+ */
+function getUserAclAllowedResources(CentreonACL $userAcl, string $resourceType)
+{
+    return match ($resourceType) {
+        'hosts' => $userAcl->getHostAclConf(null, 'broker'),
+        'servicegroups' => $userAcl->getServiceGroupAclConf(null, 'broker'),
+        'hostgroups' => $userAcl->getHostGroupAclConf(null, 'broker'),
+    };
+}
+
+/**
+ * Check resources access regarding selected resources
+ *
+ * @param CentreonACL $userAcl
+ * @param int[] $selectedResources
+ * @param string $resourceType
+ * @return bool
+ **/
+function checkResourcesRelations(CentreonACL $userAcl, array $selectedResources, string $resourceType): bool
+{
+    $allowedResources = getUserAclAllowedResources($userAcl, $resourceType);
+
+    $selectedResourceIds = array_map(
+        function (array $resource) {
+            return $resource['id'];
+        },
```

Comment:
if target > php 7.4, I suggest Arrow function here.
```suggestion
        static fn (array $resource) => $resource['id'],
```

Reply:
Yes, target branches have changed recently and the 21.10.x support has been removed from the ticket. 22.04.x already supported PHP 8.0 so 👍 for the arrow function